catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorColorVisualizeTest.kt

```diff
@@ -0,0 +1,219 @@
+
+        setTestColorWithColorPicker(0, 0)
+
+        Espresso.pressBack()
+
+        onView(withId(R.id.brick_set_color_edit_text))
+            .check(matches(ViewMatchers.withText("'#0074CD ' ")))
+
+        onView(withId(R.id.brick_set_color_edit_text))
+            .check(
+                matches(
+                    VisualizeColorMatcher(
+                        intArrayOf(getColorValueFromColorString("'#0074CD ' "))
+                    )
+                )
+            )
+    }
+
+    @Test
+    fun visualizeMultipleColorsInScriptView() {
+        onView(withId(R.id.brick_set_color_edit_text)).perform(click())
+
+        setTestColorWithColorPicker(0, 0)
+
+        onFormulaEditor().performOpenCategory(LOGIC).performSelect("or")
+
+        setTestColorWithColorPicker(2, 2)
+
+        Espresso.pressBack()
+
+        onView(withId(R.id.brick_set_color_edit_text))
+            .check(matches(ViewMatchers.withText("'#0074CD ' or '#CA0186 ' ")))
+
+        onView(withId(R.id.brick_set_color_edit_text))
+            .check(
+                matches(
+                    VisualizeColorMatcher(
+                        intArrayOf(
+                            getColorValueFromColorString("'#0074CD ' "),
+                            getColorValueFromColorString("'#CA0186 ' ")
+                        )
+                    )
+                )
+            )
+    }
+
+    @Test
+    fun visualizeColorInFormulaEditor() {
+        onView(withId(R.id.brick_set_color_edit_text))
+            .perform(click())
+
+        setTestColorWithColorPicker(0, 0)
+
+        onFormulaEditor().checkShows("'#0074CD ' ")
+
+        onView(FORMULA_EDITOR_TEXT_FIELD_MATCHER)
+            .check(
+                matches(
+                    VisualizeColorMatcher(
+                        intArrayOf(getColorValueFromColorString("'#0074CD ' "))
+                    )
+                )
+            )
+    }
+
+    @Test
+    fun visualizeMultipleColorsInFormulaEditor() {
+        onView(withId(R.id.brick_set_color_edit_text)).perform(click())
+
+        setTestColorWithColorPicker(0, 0)
+
+        onFormulaEditor().performOpenCategory(LOGIC).performSelect("or")
+
+        setTestColorWithColorPicker(2, 2)
+
+        onFormulaEditor().checkShows("'#0074CD ' or '#CA0186 ' ")
+
+        onView(FORMULA_EDITOR_TEXT_FIELD_MATCHER)
+            .check(
+                matches(
+                    VisualizeColorMatcher(
+                        intArrayOf(
+                            getColorValueFromColorString("'#0074CD ' "),
+                            getColorValueFromColorString("'#CA0186 ' ")
+                        )
+                    )
+                )
+            )
+    }
+
+    private fun setTestColorWithColorPicker(row: Int, column: Int) {
+        onView(withId(R.id.formula_editor_keyboard_color_picker))
+            .perform(click())
+
+        ColorPickerInteractionWrapper.onColorPickerPresetButton(row, column)
```

Comment:
again import properly to just call function, function name alone is sufficient to know what this is about